etl_framework/datastore_interfaces/sql_database_interface.py

```diff
@@ -167,11 +167,12 @@ def run_statement(self, sql_statement=None, new_con=False, save_con=True,
 
                 except self.CONNECTION_GONE_EXCEPTION:
                     print 'WARNING: Attempting reconnect to MySQL db. Uncommitted transactions will rollback!'
-                    con.ping(True)
-                    self._execute_statement(cursor=cursor,
-                                            sql_statement=sql_statement,
-                                            multiple_values=multiple_values,
-                                            params=params)
+                    con = self._get_connection()
+                    with closing(con.cursor()) as cursor:
```

Comment:
@ianlofs  could you also do `new_con=True` just to make it explicit that you're grabbing a new connection?